File: TrackIt.Model/Models/Ingredient.cs
namespace TrackIt.Model.Models
{
    public class Ingredient
    {
        public int IngredientId { get; set; }

        public string? Name { get; set; }

        public double? Protein { get; set; }

        public double? Fat { get; set; }

        public double? Carbs { get; set; }

        public double? Calories { get; set; }

        public virtual ICollection<MealsIngredient> MealsIngredients { get; set; } = new List<MealsIngredient>();
    }
}

File: UI/trackit_admin/lib/models/Ingredient/ingredient.dart
import 'package:json_annotation/json_annotation.dart';
part 'ingredient.g.dart';

@JsonSerializable()
class Ingredient {
  int? ingredientId;
  String? name;
  double? protein;
  double? fat;
  double? carbs;
  double? calories;
  String? image;

  Ingredient(this.ingredientId, this.fat, this.calories, this.carbs,
      this.protein, this.name, this.image);

  factory Ingredient.fromJson(Map<String, dynamic> json) =>
      _$IngredientFromJson(json);

  Map<String, dynamic> toJson() => _$IngredientToJson(this);
}
